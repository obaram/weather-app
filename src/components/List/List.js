import React from 'react';
import {Listitem} from './ListItem/Listitem'
import { Table, Alert} from 'react-bootstrap';
import styles from './List.module.scss'

const List = ({items,handleDelete,unit,error, success}) => (

    <>
    {error.show && <Alert  className={styles.alert} key="alert" variant="warning">{error.msg}</Alert>  }
    {success.show && <Alert  className={styles.alert} key="success" variant="success">{success.msg}</Alert>  } 
   
   <Table striped bordered hover className={styles.wrapper}>
    <thead>
        <tr>
        <th>#</th>
        <th>Miasto</th>
        <th>Åšrednia prognozowana tempretura</th>
        <th></th>
        </tr>
    </thead>
    <tbody>
        
        {
            items.map((item,index) => <Listitem item={item} index={index} handleDelete={handleDelete} key={index} unit={unit}/>)
        }
        
    </tbody>
    </Table>
    </>
)

export default List;
