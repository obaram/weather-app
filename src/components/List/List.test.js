import React from 'react';
import ReactDOM from 'react-dom';
import List from '../List/List';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'; 
import {BrowserRouter,Route, Switch} from 'react-router-dom'
import "../../../setupTest"

const props = {
    index: 1,
    items: [{
        city: "RzeszÃ³w",
        temp: 25.00,
        lat: 31.165,
        lon:  21.276,
    }],
    handleDelete: jest.fn(), 
    unit:'metric',
    error : {
      msg:"",
      show: false
    },  
    success : {
      msg:"",
      show:false
    }, 
} 

  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><List {...props} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });


 describe('List', () => {
     it('should render list',()=>{
     
     const wrapper = shallow(<List {...props} />);
     expect(wrapper).toMatchSnapshot();

     })

})
