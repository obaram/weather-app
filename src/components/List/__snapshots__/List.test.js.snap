// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render list item 1`] = `
<Fragment>
  <Bootstrap(Table)
    bordered={true}
    className="wrapper"
    hover={true}
    striped={true}
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          City
        </th>
        <th>
          Average temperature
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <Listitem
        handleDelete={[MockFunction]}
        index={0}
        item={
          Object {
            "city": "Rzesz처w",
            "lat": 31.165,
            "lon": 21.276,
            "temp": 25,
          }
        }
        key="0"
        unit="metric"
      />
    </tbody>
  </Bootstrap(Table)>
</Fragment>
`;

exports[`renders correctly 1`] = `
<table
  className="table wrapper table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        City
      </th>
      <th>
        Average temperature
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr
      className="wrapper"
    >
      <td>
        1
      </td>
      <td>
        <a
          href="/itemview/Rzesz처w"
          onClick={[Function]}
        >
          Rzesz처w
        </a>
      </td>
      <td>
        25
         
        <span>
          째C
        </span>
      </td>
      <td>
        <button
          className="button btn btn-outline-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-remove"
          />
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
